### PR TITLE
New Country Capital Spice

### PR DESCRIPTION
**Pull request by abhisheksp**
IA Link : https://duck.co/ia/view/country_capital

### CODE REVIEW

**Comment by daxtheduck** [comment]
@abhisheksp Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/country_capital/country_capital.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            data: {
+                country: capitalize(api_result.country),
+                capital: api_result.capital
+            },
+            meta: {
+                sourceName: "Wikipedia",
+                sourceUrl: 'https://en.wikipedia.org/w/index.php?search=' + api_result.capital
+            },
+            templates: {
+                group: 'base',
```
Comment (GuiltyDolphin) [inline share/spice/country_capital/country_capital.js]: Why are you using the `base` template? We should just be able to use `text` here.

File: lib/DDG/Spice/CountryCapital.pm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+spice is_cached => 1;
+spice proxy_cache_valid => "200 1d"; # defaults to this automatically
+
+spice wrap_jsonp_callback => 1; # only enable for non-JSONP APIs (i.e. no &callback= parameter)
+
+# API endpoint - http://docs.duckduckhack.com/walkthroughs/forum-lookup.html#api-endpoint
+spice to => 'http://countryapi.herokuapp.com/capital/$1';
+
+# Triggers - https://duck.co/duckduckhack/spice_triggers
+triggers any => 'country capital', 'what is the capital of', 'capital of', 'capital';
```
Comment (GuiltyDolphin) [inline lib/DDG/Spice/CountryCapital.pm]: We could use `startend` here.

Comment (abhisheksp) [inline lib/DDG/Spice/CountryCapital.pm]: done.

File: share/spice/country_capital/country_capital.js (hunk 1 above)
Comment (abhisheksp) [inline share/spice/country_capital/country_capital.js]: done.

**Comment by abhisheksp** [comment]
@moollaza  API source is now on OpenShift(Zero downtime).

**Comment by moollaza** [comment]
> @moollaza API source is now on OpenShift(Zero downtime).

@abhisheksp it looks like OpenShift free plans do idle as well, but it's not forced. However if it goes 24 hours without a request it will be "paused". 

I just realized though that this is all static data -- I don't think it's necessary to use an API because we can put all this into a YAML file, for example we could copy the info from Wiki here: https://en.wikipedia.org/wiki/List_of_national_capitals_in_alphabetical_order or even scrape it into a Fathead

**Comment by abhisheksp** [comment]
I am currently working on the extension of the api to return generic thumbnail of the given capital.
Should I still be working on converting this to a fathead?

**Comment by GuiltyDolphin** [comment]
@moollaza @abhisheksp Perhaps we could have a display similar to the Wikipedia FatHead? If you queried `capital city of England`, for example, you could have:

screenshot from 2016-04-20 14 04 47

This clearly answers the query (`London` is the title), and displays some useful additional information people may be looking for.

So we could use the list you linked to from Wikipedia, and use that as an additional lookup? Or perhaps have a new FatHead parse the query and then call the Wikipedia FatHead.

**Comment by moollaza** [comment]
@abhisheksp @GuiltyDolphin I've made a task for our team to investigate. It looks like some queries already provide a Wiki result (https://duckduckgo.com/?q=capital+of+canada&ia=about) because Wikipedia has a redirect for that query to the page for the capital city. However, not all "Capital of X" queries on WIki react like that: https://duckduckgo.com/?q=capital+of+india

It looks like there are a few that _do_ redirect, however we're not showing the correct result: https://en.wikipedia.org/wiki/Capital_of_australia

We're going to investigate this and see if we can better handle these queries internally first.

**Comment by abhisheksp** [comment]
Any updates on this?

**Comment by pjhampton** [comment]
Hey @abhisheksp 

[DuckDuckHack is now in Maintenance Mode](https://duckduckhack.com/) and from now on, we are only accepting issues and PRs for essential bugs and bug fixes.

Unfortunately, this is non-essential and will be closed.

We appreciate you taking the time to contribute and apologize for not being able to include your work.

**Pull request closed without merging**